maro/rl/models/learning_model.py

```diff
@@ -1,37 +1,44 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT license.
 
-import torch
 import torch.nn as nn
 
 
-class IdentityLayers(nn.Module):
-    """A convenience dummy model that effects the identity transformation."""
-    def forward(self, inputs):
-        return inputs
-
-
 class LearningModel(nn.Module):
-    """A general model abstraction that consists of representation layers and decision layers.
+    """NN model that consists of multiple shared blocks and multiple heads.
 
-    Args:
-        representation_layers (nn.Module): An NN-based feature extractor.
-        decision_layers (nn.Module): An NN model that takes the output of the representation layers as input
-            and outputs values of interest in RL (e.g., state & action values).
-        clip_value (float): Threshold used to clip gradients.
+    The shared blocks must be chainable, i.e., the output dimension of a block must match the input dimension of
+    its successor. Heads must be provided in the form of keyword arguments. If at least one head is provided, the
+    output of the model will be a dictionary with the names of the heads as keys and the corresponding head outputs
+    as values. Otherwise, the output will be the output of the last block.
     """
-    def __init__(
-        self,
-        representation_layers: nn.Module = IdentityLayers(),
-        decision_layers: nn.Module = IdentityLayers(),
-        clip_value: float = None
-    ):
+    def __init__(self, *blocks, **heads):
         super().__init__()
-        self._net = nn.Sequential(representation_layers, decision_layers)
-
-        if clip_value is not None:
-            for param in self._net.parameters():
-                param.register_hook(lambda grad: torch.clamp(grad, -clip_value, clip_value))
-
-    def forward(self, inputs):
-        return self._net(inputs)
+        self._shared = nn.Sequential(*blocks)
+        self._heads = heads
```

Comment:
task_heads

Reply:
task_nets which will use nnSequential to combine representation_stack and task_head